src/CommunityToolkit/Xamarin.CommunityToolkit/Views/DrawingView/Extensions/Extensions.shared.cs

```diff
@@ -15,26 +15,28 @@ public static class Extensions
 		public static ObservableCollection<Point> SmoothedPathWithGranularity(this ObservableCollection<Point> currentPoints,
 			int granularity)
 		{
+			var currentPointsCopy = new ObservableCollection<Point>(currentPoints);
+
 			// not enough points to smooth effectively, so return the original path and points.
-			if (currentPoints.Count < granularity + 2)
-				return new ObservableCollection<Point>(currentPoints);
+			if (currentPointsCopy.Count < granularity + 2)
+				return currentPointsCopy;
 
 			var smoothedPoints = new ObservableCollection<Point>();
 
 			// duplicate the first and last points as control points.
-			currentPoints.Insert(0, currentPoints[0]);
-			currentPoints.Add(currentPoints.Last());
+			currentPointsCopy.Insert(0, currentPointsCopy[0]);
+			currentPointsCopy.Add(currentPointsCopy.Last());
```

Comment:
can we avoid the usage of Linq here?

Reply:
originally we used Index here, but it is not available in netstandard1.0. replaced with [ ]